manticore/ethereum.py

```diff
@@ -287,21 +287,100 @@ def __mul__(self, reps):
+                    raise Value("value '%s' is too large to be an address" % hex(value))
+                self.value = value
+            elif isinstance(value, str) and value[:2] == '0x':
+                if len(value) != 42:
+                    raise ValueError("hex '%s' does not have 'address'-length" % value)
+                self.value = long(value, 16)
+            elif isinstance(value, tuple):
+                if len(value) != 20:
+                    raise ValueError("value %s does not have 'address'-length" % (value, size))
+                self.value = ABI._tuple_bytes_to_long(value)
+            else:
+                raise TypeError('Expected int, long, str, or tuple')
+
+        def serialize(self):
+            return ABI.serialize_uint(self.value)
+
+    class ExternalFunction(object):
+        def __init__(self, address, selector):
+            if not isinstance(address, ABI.Address):
+                raise TypeError("Expected ABI.Address, found %s" % type(address))
+            if not isinstance(selector, ABI.StaticBytes):
+                raise TypeError("Expected ABI.StaticBytes, found %s" % type(selector))
+            if selector.size() != 4:
+                raise ValueError("Expected selector size of 4, found %s", selector.size())
+            self.address = address
+            self.selector = selector
+
+        def serialize(self):
+            return (
+                ABI.serialize_uint(self.address.value, 20) +
+                self.selector.sub_word +
+                tuple('\0' * 8)
+            )
+
     @staticmethod
     def serialize(value):
         '''
         Translates a Python object to its EVM ABI serialization.
         '''
         if isinstance(value, (str, tuple)):
-            return ABI.serialize_string(value)
-        if isinstance(value, (list)):
-            return ABI.serialize_array(value)
-        if isinstance(value, (int, long)):
-            return ABI.serialize_uint(value)
-        if isinstance(value, ABI.SByte):
-            return ABI.serialize_uint(value.size) + (None,) * value.size + (('\x00',) * (32 - (value.size % 32)))
-        if value is None:
-            return (None,) * 32
+            result = ABI.serialize_string(value)
+        elif isinstance(value, (list)):
+            result = ABI.serialize_array(value)
+        elif isinstance(value, (int, long)):
+            result = ABI.serialize_uint(value)
+        elif isinstance(value, ABI.StaticBytes):
+            result = value.serialize()
+        elif isinstance(value, ABI.Address):
+            result = value.serialize()
+        elif isinstance(value, ABI.ExternalFunction):
+            result = value.serialize()
+        elif isinstance(value, ABI.SByte):
+            result = ABI.serialize_uint(value.size) + (None,) * value.size + (('\x00',) * (32 - (value.size % 32)))
+        elif value is None:
+            result = (None,) * 32
+        else:
+            raise TypeError
```

Comment:
let's add a descriptive error message here expressing that the type of value is not supported

Reply:
fixed in https://github.com/trailofbits/manticore/pull/875/commits/63efc4fb384bdd36f3e54d614b1c5e5625098f14